test/utest_ddimm_parser.cpp

```diff
@@ -0,0 +1,118 @@
+#include "types.hpp"
+
+#include <exception>
+#include <fstream>
+
+#include <gtest/gtest.h>
+
+using namespace vpd;
+
+class DdimmVpdParserTest : public ::testing::Test
+{
+  protected:
+    types::BinaryVector m_ddimmVpdVector;
+
+    DdimmVpdParserTest()
+    {
+        // Open the DDIMM DDR5 VPD file in binary mode
+        std::ifstream l_kwVpdFile("ddr5_ddimm.dat", std::ios::binary);
+
+        // Read the content of the binary file into a vector
+        m_ddimmVpdVector.assign((std::istreambuf_iterator<char>(l_kwVpdFile)),
+                                std::istreambuf_iterator<char>());
+    }
+};
+
+TEST_F(DdimmVpdParserTest, GoodTestCase)
+{
+    types::DdimmVpdMap l_ddimmMap{
+        std::pair<std::string, size_t>{"MemorySizeInKB", 0x4000000},
+        std::pair<std::string, types::BinaryVector>{
+            "FN", {0x30, 0x33, 0x48, 0x44, 0x37, 0x30, 0x30}},
+        std::pair<std::string, types::BinaryVector>{
+            "PN", {0x30, 0x33, 0x48, 0x44, 0x37, 0x30, 0x30}},
+        std::pair<std::string, types::BinaryVector>{"SN",
+                                                    {0x59, 0x48, 0x33, 0x33,
+                                                     0x31, 0x54, 0x33, 0x38,
+                                                     0x34, 0x30, 0x33, 0x46}},
+        std::pair<std::string, types::BinaryVector>{"CC",
+                                                    {0x33, 0x32, 0x41, 0x31}}};
+
+    DdimmVpdParser l_ddimmParser(std::move(m_ddimmVpdVector));
+    auto l_parsedDdimmMap =
+        std::move(get<types::DdimmVpdMap>(l_ddimmParser.parse()));
+
+    ASSERT_EQ(1, l_ddimmMap == l_parsedDdimmMap);
+}
+
+TEST_F(DdimmVpdParserTest, InvalidDdrType)
+{
+    // Invalid DDR type, corrupted at index[2]
+    m_ddimmVpdVector[constants::SPD_BYTE_2] = 0x13;
+    DdimmVpdParser l_ddimmParser(std::move(m_ddimmVpdVector));
+
+    EXPECT_THROW(l_ddimmParser.parse(), std::exception);
+}
+
+TEST_F(DdimmVpdParserTest, ZeroDdimmSize)
+{
+    // Badly formed DDIMM VPD data - corrupted at index[235],
+    // ddimm size calculated a zero
+    m_ddimmVpdVector[constants::SPD_BYTE_235] = 0;
+    DdimmVpdParser l_ddimmParser(std::move(m_ddimmVpdVector));
+
+    EXPECT_THROW(l_ddimmParser.parse(), std::exception);
+}
+
+TEST_F(DdimmVpdParserTest, InvalidValue)
+{
+    // Out of range data, fails to check valid value - corrupted at index[4]
+    m_ddimmVpdVector[constants::VALUE_4] = 0;
+    DdimmVpdParser l_ddimmParser(std::move(m_ddimmVpdVector));
+
+    EXPECT_THROW(l_ddimmParser.parse(), std::exception);
+}
+
+TEST_F(DdimmVpdParserTest, EmptyInputVector)
+{
+    // VPD is empty
+    types::BinaryVector l_emptyVector{};
+    DdimmVpdParser l_ddimmParser(l_emptyVector);
+
+    EXPECT_THROW(l_ddimmParser.parse(), std::exception);
+}
+
+TEST_F(DdimmVpdParserTest, InvalidVpdType)
+{
+    // Invalid VPD type - corrupted at index[3]
+    m_ddimmVpdVector[constants::SPD_BYTE_3] = constants::SPD_MODULE_TYPE_DDIMM +
```

Comment:
What is the point of having the file is vector that is being passed is hardcoded?

Reply:
Updated the test case.